diff --git a/README.rst b/README.rst
index 26266a9..c00376c 100644
--- a/README.rst
+++ b/README.rst
@@ -26,6 +26,7 @@ Dependencies
 
 Documentation
 -------------
+
 * http://petsc4py.readthedocs.org/, This does not contain the epydoc-generated API reference.
 
-* https://pythonhosted.org/petsc4py/, This is for the last release, not the in-development version.
+* http://www.mcs.anl.gov/petsc/petsc4py-current/docs/, This is for the last release, not the in-development version.
diff --git a/conf/baseconf.py b/conf/baseconf.py
index 1a8cab0..3333170 100644
--- a/conf/baseconf.py
+++ b/conf/baseconf.py
@@ -110,6 +110,9 @@ class PetscConfig:
     def __getitem__(self, item):
         return self.configdict[item]
 
+    def get(self, item, default=None):
+        return self.configdict.get(item, default)
+
     def configure(self, extension, compiler=None):
         self.configure_extension(extension)
         if compiler is not None:
@@ -197,17 +200,19 @@ class PetscConfig:
 
     def configure_compiler(self, compiler):
         if compiler.compiler_type != 'unix': return
-        (cc, cxx, cflags, ccshared,
-         ldflags, ldshared, so_ext) = get_config_vars(
-            'CC', 'CXX', 'CFLAGS',  'CCSHARED',
-            'LDFLAGS', 'LDSHARED', 'SO')
+        # distutils C/C++ compiler
+        (cc, cflags, ccshared, cxx) = get_config_vars(
+            'CC', 'CFLAGS',  'CCSHARED', 'CXX')
         cflags = cflags or ''
-        ldflags = ldflags or ''
         cflags = cflags.replace('-Wstrict-prototypes', '')
+        ccshared = ccshared or ''
+        # distutils linker
+        (ldflags, ldshared, so_ext) = get_config_vars(
+            'LDFLAGS', 'LDSHARED', 'SO')
         ld = cc
-        ldshared = ldshared.replace(ld, '', 1).strip()
-        ldshared = [flg for flg in split_quoted(ldshared)
-                    if flg not in split_quoted(ldflags)]
+        ldflags = ldflags or ''
+        ldcmd = split_quoted(ld) + split_quoted(ldflags)
+        ldshared = [flg for flg in split_quoted(ldshared) if flg not in ldcmd]
         ldshared = str.join(' ', ldshared)
         #
         getenv = os.environ.get
@@ -218,20 +223,12 @@ class PetscConfig:
         PCC = self['PCC']
         PCC_FLAGS = get_flags(cc) + ' ' + self['PCC_FLAGS']
         PCC_FLAGS = PCC_FLAGS.replace('-fvisibility=hidden', '')
-        if sys.version_info[:2] < (2, 5):
-            PCC_FLAGS = PCC_FLAGS.replace('-Wwrite-strings', '')
         PCC = getenv('PCC', PCC) + ' ' +  getenv('PCCFLAGS', PCC_FLAGS)
         ccshared = getenv('CCSHARED', ccshared)
         cflags   = getenv('CFLAGS',   cflags)
         PCC_SHARED = str.join(' ', (PCC, ccshared, cflags))
         # C++ compiler
-        if self.language == 'c++':
-            PCXX = PCC
-        else:
-            try:
-                PCXX = self['CXX']
-            except KeyError:
-                PCXX = cxx
+        PCXX = PCC if self.language == 'c++' else self.get('CXX', cxx)
         # linker
         PLD = self['PCC_LINKER']
         PLD_FLAGS = get_flags(ld) + ' ' + self['PCC_LINKER_FLAGS']
@@ -603,6 +600,23 @@ class sdist(_sdist):
 
 # --------------------------------------------------------------------
 
+if setuptools:
+    try:
+        from setuptools.command import egg_info as mod_egg_info
+        _FileList = mod_egg_info.FileList
+        class FileList(_FileList):
+            def process_template_line(self, line):
+                level = log.set_threshold(log.ERROR)
+                try:
+                    _FileList.process_template_line(self, line)
+                finally:
+                    log.set_threshold(level)
+        mod_egg_info.FileList = FileList
+    except:
+        pass
+
+# --------------------------------------------------------------------
+
 def append(seq, item):
     if item not in seq:
         seq.append(item)
diff --git a/src/PETSc.c b/src/PETSc.c
index 108afa0..5a54aa1 100644
--- a/src/PETSc.c
+++ b/src/PETSc.c
@@ -1,3 +1,4 @@
 #define MPICH_SKIP_MPICXX 1
 #define OMPI_SKIP_MPICXX 1
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
 #include "petsc4py.PETSc.c"
diff --git a/src/PETSc.py b/src/PETSc.py
index e9ad1b4..69fcb3c 100644
--- a/src/PETSc.py
+++ b/src/PETSc.py
@@ -1,3 +1,6 @@
+import warnings
+warnings.filterwarnings("ignore", message="numpy.dtype size changed")
+warnings.filterwarnings("ignore", message="numpy.ndarray size changed")
 ARCH = None
 from petsc4py.lib import ImportPETSc
 PETSc = ImportPETSc(ARCH)
